test/macro.test.ts

```diff
@@ -178,4 +179,39 @@ suite('Record and execute a macro', () => {
     end: ['one t|wo three'],
     statusBar: 'E748: No previously used register',
   });
+
+  suite('Text copied into register can be run as a macro', () => {
+    const start = ['one', 'two', 'three'];
+    const register = 'x';
+    const testCases: Array<[string, string[]]> = [
+      ['', ['|one', 'two', 'three']],
+      ['j', ['one', '|two', 'three']],
+      ['2j', ['one', 'two', '|three']],
+
+      ['A' + ', uno', ['one, un|o', 'two', 'three']],
+
+      ['dd', ['|two', 'three']],
+
+      ['yyp', ['one', '|one', 'two', 'three']],
+
+      ['VGJ', ['one two| three']],
+
+      ['gUU' + 'j.' + 'j.', ['|ONE', 'TWO', 'THREE']],
+
+      [':2d', ['|one', 'three']],
```

Comment:
This one should include an `<Enter>` at the end, this is the behavior in Vim also.  Try `:2d` vs `:2d\n`. Only the latter works.

I've tried with `<Enter>` and with `\n` but couldn't get it to work (yet). It's interesting, since in the remap config I have something like this that works. I'll have to see why it works there and not here.
```json
  "vim.normalModeKeyBindingsNonRecursive": [
        {
            "after": [
                ":",
                "s",
                "/",
                "\\",
                "\\",
                "/",
                "\\",
                "/",
                "/",
                "g",
                "<Enter>"
            ],
            "before": [
                "<leader>",
                "s",
                "/"
            ]
        },
```